### PR TITLE
add support for gyro pools

### PR DESCRIPTION
**Pull request by groninge01**
- add support (join/exit/some details) for gyroscope pools
- disable custom invest / single asset withdraw for gyroscope pools
- add tooltip for disabled choice
- make 'Recommended' text transparant when it's the only choice
- disable 'Withdraw' on pool detail when balance in pool is 0
- add info toast for the Gyroscope ECLP USDT/USDC pool
- add 'built-by' badge for the Gyroscope ECLP USDT/USDC pool on pool detail page
- above also on pool list page (assumption: boosted-by & built-by don't apply at the same time)

TODO: 
- add a better 'built-by-gyroscope' image for the badge :)
- merge backend canary to production

### CODE REVIEW

**Comment by danielmkm** [comment]
Why is this failing to build?

File: lib/services/pool/pool-gyro.service.ts [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+        tokenAmountsIn: TokenAmountHumanReadable[],
+        slippage: AmountHumanReadable,
+    ): Promise<PoolJoinEstimateOutput> {
+        const bptAmount = poolWeightedExactTokensInForBPTOut(tokenAmountsIn, this.pool);
+
+        if (bptAmount.lt(0)) {
+            return { priceImpact: 0, minBptReceived: '0' };
+        }
+
+        const bptZeroPriceImpact = poolWeightedBptForTokensZeroPriceImpact(tokenAmountsIn, this.pool);
```
Comment (danielmkm) [inline lib/services/pool/pool-gyro.service.ts:100]: price impact is always zero, this isn't needed

File: lib/services/pool/lib/util.ts [hunk 2]
```diff
@@ -235,11 +251,12 @@ export function poolWeightedExactTokensInForBPTOut(
 
 export function poolWeightedBptForTokensZeroPriceImpact(
     tokenAmounts: TokenAmountHumanReadable[],
-    pool: GqlPoolWeighted,
+    pool: GqlPoolWeighted | GqlPoolGyro,
```
Comment (danielmkm) [inline lib/services/pool/lib/util.ts:254]: this isn't needed, price impact is always 0 for gryo pools

File: modules/pool/invest/components/PoolInvestTypeChoice.tsx [hunk 3]
```diff
@@ -22,7 +22,7 @@ interface Props {
 }
 
 export function PoolInvestTypeChoice({ onShowProportional, onShowCustom }: Props) {
-    const { pool, poolService } = usePool();
+    const { pool, poolService, isGyroPool } = usePool();
```
Comment (danielmkm) [inline modules/pool/invest/components/PoolInvestTypeChoice.tsx:25]: bad pattern sir, plz no. Make this a generic concept

**Review by danielmkm — CHANGES_REQUESTED** [review]

**Comment by groninge01** [comment]
> Why is this failing to build?

Because it needs changes from the canary backend which aren't yet in production.

File: components/inputs/BeetsTokenInputWithSlider.tsx [hunk 4]
```diff
@@ -168,7 +168,7 @@ export function BeetsTokenInputWithSlider({
                             cursor={hasBalance ? 'pointer' : 'default'}
                         >
                             You have {tokenFormatAmount(balance)}
-                            {hasBalance ? (
+                            {hasBalance && selectedTokenOption.address !== networkConfig.eth.address.toLowerCase() ? (
```
[the remark this answers is not in the document]
Comment (groninge01) [reply components/inputs/BeetsTokenInputWithSlider.tsx:171]: It is. Removing it here.

File: components/built-badge/lib/built-by-tooltips.ts [hunk 5]
```diff
@@ -0,0 +1,3 @@
+export const BuiltByTooltips: { [type: string]: string } = {
```
[the remark this answers is not in the document]
Comment (groninge01) [reply components/built-badge/lib/built-by-tooltips.ts]: Made it generic.

File: lib/services/pool/pool-util.ts [hunk 6]
```diff
@@ -104,6 +106,8 @@ export function poolGetTypeName(pool: GqlPoolUnion) {
             return 'Liquidity bootstrapping pool';
         case 'GqlPoolMetaStable':
             return 'MetaStable pool';
+        case 'GqlPoolGyro':
+            return 'Gyro pool';
```
[the remark this answers is not in the document]
Comment (groninge01) [reply lib/services/pool/pool-util.ts]: Updated this.

File: modules/pool/detail/components/warning/PoolWarningThirdParty.tsx [hunk 7]
```diff
@@ -0,0 +1,26 @@
+import { Box, HStack, Link } from '@chakra-ui/react';
```
[the remark this answers is not in the document]
Comment (groninge01) [reply modules/pool/detail/components/warning/PoolWarningThirdParty.tsx]: Removed popup, will add the banner image as soon as it's ready.

**Review by danielmkm — APPROVED** [review]

**Pull request merged**